Homework6/pingpong.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <chrono>
+#include <atomic>
+#include <mutex>
+#include <thread>
+#include <condition_variable>
+
+using namespace std;
+
+constexpr chrono::milliseconds flightTime = 300ms;
+constexpr chrono::milliseconds clockTick = 200ms;
+
+enum gameState {
+    FINISH = 0, PING, PONG
+};
+
+class PingPong {
+    mutex ballLock;
+
+    atomic<int> counter;
+    atomic<int> turnNumber = 1;
+    atomic<bool> timerOut = false;
+
+    atomic<gameState> status = gameState::PING;
+
+    condition_variable condition;
+public:
+
+    explicit PingPong(int turns) : counter(turns) {}
+
+    void ping() {
+        while (counter && !timerOut) {
+            unique_lock<mutex> lock(ballLock);
+            condition.wait_for(lock, clockTick, [&]{return !timerOut;});
```

Comment:
dalej jest tutaj wait_for, przez co czasami duże opóźnienia są na timeoucie (nawet o sekundę) i można to zrobić ze zwykłym wait ale w sumie wszystko działa poprawnie :)